File: Java Programs/5.0.4 War.java
/*
 # - - - - - - - - - - - - - - - - #
 # SummerTech/Coditum Curriculum   #
 # Created by Gabe Ehrlich         #
 #                                 #
 # Report any bugs, issues, or     #
 # corrections to:                 #
 # matt@summertech.net             #
 #                                 #
 #                                 #
 # Module 5.0.4                    #
 # - - - - - - - - - - - - - - - - #
 
 SUMMERTECH CONFIDENTIAL
 COPYRIGHT (C) 2017 SUMMERTECH, INC. ALL RIGHTS RESERVED
 */


import java.util.ArrayList;

public class War {
  
  public static void main(String[] args) throws InterruptedException {
    
    //Variables
    Deck d = new Deck();
    d.shuffle();
    ArrayList<Card> p1 = new ArrayList<Card>();
    ArrayList<Card> p2 = new ArrayList<Card>();
    ArrayList<Card> playing = new ArrayList<Card>();
    int warCount = 0;
    boolean game = true;
    
    //Split the deck into two
    while(d.getSize() > 0) {
      p1.add(d.draw());
      p2.add(d.draw());
    }
    
    //Game loop
    while(game == true) {
      
      //Announce then Play cards
      System.out.println("Player one flips: " + p1.get(0) + "   Player two flips: " + p2.get(0));
      System.out.println("Player one size: " + p1.size()+"   Player two size: " + p2.size() +
                         "    Playing field size: " + playing.size());
      Thread.sleep(500);
      playing.add(0, p1.get(0));
      p1.remove(0);
      playing.add(1, p2.get(0));
      p2.remove(0);
      
      //Comparing the two most recent cards
      //Player one wins
      if(playing.get(0).getValue() > playing.get(1).getValue()) {
        System.out.println("Player one wins the hand.");
        while(!playing.isEmpty()) {
          p1.add(playing.get(0));
          playing.remove(0);
        }
      }
      //Player two wins
      else if(playing.get(0).getValue() < playing.get(1).getValue()) {
        System.out.println("Player two wins the hand.");
        while(!playing.isEmpty()) {
          p2.add(playing.get(0));
          playing.remove(0);
        }
      }
      //The only remaining case is war
      else {
        System.out.println("A WAR HAS OCCURRED.");
        Thread.sleep(1500);
        //Makes sure that a war in the last few cards won't just crash the game.
        if(p1.size() > 4 && p2.size() > 4) {
          for(int i = 0; i < 3; i++) {
            playing.add(p1.get(0));
            p1.remove(0);
            playing.add(p2.get(0));
            p2.remove(0);
          }
        }
        //Should a player not have the required three bonus cards, it will put in all but one.
        else {
          while(p1.size() != 1 && p2.size() != 1) {
            playing.add(p1.get(0));
            p1.remove(0);
            playing.add(p2.get(0));
            p2.remove(0);
          }
        }
      }
      
      //Win conditions
      if(p1.size() == 52 && playing.isEmpty()) {
        System.out.println("Player one wins!");
        game = false;
      }
      else if(p2.size() == 52 && playing.isEmpty()) {
        System.out.println("Player two wins!");
        game = false;
      }
    }
  }
  
}

File: Python Programs/5.0.4 War.py
# - - - - - - - - - - - - - - - - #
# SummerTech/Coditum Curriculum   #
# Created by Matthew Baptist      #
#                                 #
# Report any bugs, issues, or     #
# corrections to:                 #
# matt@summertech.net             #
#                                 #
#                                 #
# Module 5.0.4                    #
# - - - - - - - - - - - - - - - - #

#SUMMERTECH CONFIDENTIAL
#COPYRIGHT (C) 2017 SUMMERTECH, INC. ALL RIGHTS RESERVED

from Deck import Deck

deck = Deck()
deck.shuffle()

playerCard = None
player2Card = None

player1 = input("Player 1, enter your name. ")
player2 = input("Player 2, enter your name. ")

score = 0
score2 = 0

while (deck.getLength() > 0):
  player1Card = deck.draw()
  player2Card = deck.draw()
  
  print(player1 + ":", player1Card.toString())
  print(player2 + ":", player2Card.toString())
  
  if (player1Card.value > player2Card.value):
    print(player1, "wins the round.")
    score += 1
  elif (player1Card.value == player2Card.value):
    print("Tie round.")
  else:
    print(player2, "wins the round")
    score2 += 1

  input("Press enter to start the next round. ")

if (score > score2):
  print(player1, "wins with a score of", score)
elif (score == score2):
  print("Tied game!")
else:
  print(player2, "wins with a score of", score2)
